py/objarray.c

```diff
@@ -430,7 +430,7 @@ STATIC mp_obj_t array_subscr(mp_obj_t self_in, mp_obj_t index_in, mp_obj_t value
                 }
 
                 // TODO: check src/dst compat
-                mp_int_t len_adj = src_len - (slice.stop - slice.start);
+                long len_adj = src_len - (slice.stop - slice.start);
```

Comment:
I'm not sure making it `long` is the right thing to do... it's not obvious what it should be but how about just keeping it as `mp_int_t` because (so far) that seems to work?  Otherwise I'd suggest using `ssize_t`.

Reply:
I tried `ssize_t`, but that is a posix type, not a standard C type, so it causes issues on some ports with the type not being defined. So yes, we can keep it as `mp_int_t`, but will have to add more casts elsewhere.

Reply:
I tested this out building the coverage build with clang on Linux and I get the same warnings, and can fix this one simply by adding the size_t cast below, I didn't need to change the type here (it could remain `mp_int_t`).  So I'd prefer that, to not change this type.